crates/modelardb_manager/src/cluster.rs

```diff
@@ -113,6 +128,21 @@ impl Cluster {
         }
     }
 
+    /// Return the cloud node in the cluster that is currently most capable of running a query.
+    /// If there are no cloud nodes in the cluster, return [`ClusterError`](ModelarDbError::ClusterError).
+    pub fn query_node(&mut self) -> Result<Node, ModelarDbError> {
+        if let Some(query_node) = self.query_queue.pop_front() {
```

Comment:
The decision was made to simply clone the nodes since they are extremely cheap structs with very little data. Using indices would create some annoying small problems such as how to handle when a node is removed so it seemed to be overkill for such as simple implementation. 

If we want to make the nodes more complex later when looking into making the decision of choosing the query node more complex, then it might be relevant to change this.